src/include/misc.i

```diff
@@ -38,17 +54,16 @@ __wt_rdtsc(WT_SESSION_IMPL *session) {
 #if (defined __i386)
 	uint64_t x;
 
-	WT_UNUSED(session);
-
 	__asm__ volatile ("rdtsc" : "=A" (x));
+	__tsc_check_monotonic(session, &x);
 	return (x);
 #elif (defined __amd64)
-	uint64_t a, d;
-
-	WT_UNUSED(session);
+	uint64_t a, d, x;
 
 	__asm__ volatile ("rdtsc" : "=a" (a), "=d" (d));
-	return ((d << 32) | a);
+	x = (d << 32) | a;
+	__tsc_check_monotonic(session, &x);
+	return (x);
```

Comment:
I noticed `__wt_rdtsc()` has compiler-specific assembly code, which has historically been put into `include/gcc.h` or `include/msvc.h`, but this particular assembly code is shared between gcc/clang and MSVC.

There's also x86 #ifdef's in `__conn_calibrate_ticks()` in `conn_api.c`, which bothers me, we don't want architecture specific code climbing up into "portable" code.

@michaelcahill, @agorrod, any thoughts on where x86-specific code like this should land?